src/tools/storybook-addon-code/components/CodeToggle.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { IconButton } from '@storybook/components';
+import { EVENTS } from '../constants';
+
+export class CodeToggle extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      expanded: false,
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    const { channel } = this.props;
+    const { expanded } = this.state;
+    const newState = !expanded;
+
+    this.setState({
+      expanded: newState,
+    });
+
+    channel.emit(EVENTS.TOGGLE_CODE, newState);
+  }
+
+  render() {
+    const { expanded } = this.state;
+
+    return (
+      <IconButton
+        key="code"
+        active={!!expanded}
+        title={!expanded ? 'Show source code' : 'Hide source code'}
+        onClick={this.handleClick}
+      >
+        Prototype with the source code
```

Comment:
Why replacing with such a long label? Isn’t it better to keep previous title `HTML`

Reply:
As discussed, might it make more sense to have the HTML tabs in the Addons section next to TWIG, CSS, ... tabs?

Reply:
I'm ok with any solution desired by the team. The one in this pull request is the existing one from v2. If there is an alignment that it's to be put in the other panel there will be a need of a rewrite. Which I don't mind

Reply:
This one I managed to do together with the panel repositioning.